db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/fk/Graph.java

```diff
@@ -71,4 +132,50 @@ public int getVertexCount() {
         return vertexIndex.size();
 
     }
+
+    public EdgeNode.Type getEdgeType(int from, int to) {
+        EdgeNode edgeNode = getEdge(from);
+        while(true) {
+            if(edgeNode == null) {
+                throw new IllegalArgumentException("no edge between " + getName(from) + " and " + getName(to));
+            }
+            if(edgeNode.y == to) {
+                return edgeNode.type;
+            }
+            edgeNode = edgeNode.next;
+        }
+    }
+
+    public void removeEdge(int from, int to) {
+        EdgeNode edgeNode = getEdge(from);
+        if(edgeNode == null) {
+            throw new IllegalArgumentException("no edge between " + getName(from) + " and " + getName(to));
+        }
```

Comment:
Is this an error that we expected? I mean, talking about just an abstract graph, if we want to delete an edge that doesn't exist, might just be a no-op and OK to be safely ignored. Or if we don't expect this ever happen, probably assert instead of throw.

Reply:
You're right, it should never happen, but still I would rather keep an exception to get more context if something goes wrong.